### PR TITLE
feat(synthetics): Synthetics Legacy Runtime EOL final phase

### PR DESCRIPTION
**Pull request by pranav-new-relic**
### Summary

As an extension of the first phase of the Synthetics Legacy Runtime EOL that was performed on August 26, 2024 (in the Terraform Provider via #2731), this PR brings about **changes for the final phase of the Synthetics Legacy Runtime EOL** which shall go live on October 23, 2024, in order to reflect validation changes made in the Synthetics API published on the same day, in the Terraform Provider.

The changes have been rigorously tested with all possible scenarios: empty runtime, no runtime, usage of `use_unsupported_legacy_runtime` with these cases, with both update and create, with all six monitors affected by this EOL.

### Changes

This PR includes the following changes
- An amendment to the custom validator method written against all synthetic monitor resources to prevent usage of the legacy runtime with new monitors, to also extend this to existing monitors, in lieu of the final phase, according to which no monitor would be allowed to use the legacy runtime any longer
- Updates to relevant tests, removal of references of the legacy runtime from obsolete documentation
- Updates to the documentation pages of Synthetic Monitor resources with relevant warnings and information that the EOL has taken place; along with updates to the migration guide of the Synthetics Legacy Runtime EOL to help customers understand action to be taken post the EOL if they have not migrated to the new runtime yet.

### CODE REVIEW

File: website/docs/r/synthetics_broken_links_monitor.html.markdown [hunk 1]
```diff
@@ -42,16 +44,17 @@ The following are the arguments supported by this resource.
...
-
 * `tag` - (Optional) The tags that will be associated with the monitor. See [Nested tag blocks](#nested-tag-blocks) below for details
+* `runtime_type` - (Optional) The runtime that the monitor will use to run jobs (`NODE_API`).
```
Comment (shashank-reddy-nr) [inline website/docs/r/synthetics_broken_links_monitor.html.markdown:48]: I feel runtime_type, runtime_type fields are mandatory now right? If yes, optional can be removed and required can be added

**Comment by shashank-reddy-nr** [comment]
Changes looks good to me

File: website/docs/r/synthetics_broken_links_monitor.html.markdown (hunk 1 above)
Comment (pranav-new-relic) [reply website/docs/r/synthetics_broken_links_monitor.html.markdown:48]: explained in a different channel

**Comment by shashank-reddy-nr** [comment]
Approved

**Pull request merged**